src/app/public/modules/date-pipe/date.pipe.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  DatePipe
+} from '@angular/common';
+
+import {
+  Pipe,
+  PipeTransform
+} from '@angular/core';
+
+import {
+  SkyAppLocaleProvider
+} from '@skyux/i18n';
+
+import {
+  BehaviorSubject
+} from 'rxjs/BehaviorSubject';
+
+@Pipe({
+  name: 'skyDate',
+  pure: false
+})
+export class SkyDatePipe implements PipeTransform {
+  private get format(): string {
+    return this._format || 'short';
+  }
+
+  private set format(value: string) {
+    if (value && value !== this._format) {
+      this._format = value;
+    }
+  }
+
+  private get locale(): string {
+    return this._locale;
+  }
+
+  private set locale(value: string) {
+    if (value && value !== this._locale) {
+      this._locale = value;
+
+      // Create a new pipe when the locale changes.
+      this.ngDatePipe = new DatePipe(this.locale);
+    }
+  }
+
+  private defaultLocale = 'en-US';
+  private ngDatePipe: DatePipe;
+  private transformStream = new BehaviorSubject<string>('');
+  private value: Date;
+
+  private _format: string;
+  private _locale: string;
+
+  constructor(
+    private localeProvider: SkyAppLocaleProvider
+  ) {
+    this.locale = this.defaultLocale;
+
+    this.localeProvider.getLocaleInfo()
+      .subscribe((localeInfo) => {
```

Comment:
Should we have an `.takeUntil(this.ngUnsubscribe)` here?

Reply:
We can't, because pipe's do not recognize `ngOnDestroy`. Since this pipe is asynchronous, it requires that the user ALSO use Angular's `async` pipe. The `async` pipe automatically handles observable completion.

Reply:
I don't think we want the consumer to have to chain this pipe with the `async` pipe.  Instead, I would expect this pipe to be an impure pipe, similar to the `async` and the `skyAppResources` pipes. https://angular.io/guide/pipes#impure-pipes

Reply:
@Blackbaud-PaulCrowder You're correct, I missed a vital piece to this. Changes made!